xpk/README.md

```diff
@@ -106,6 +106,53 @@ cleanup with a `Cluster Delete`.
     --cluster xpk-test
     ```
 
+# How to add docker images to a xpk workload
+
+The default behavior is that the local running directory will be built into
+a docker image by xpk and those files are accessable by `xpk workload create`.
+See below for the two ways that xpk supports configuring docker images for the
+workload. Do not mix arguments from these two ways like `--base-docker-image`
+and `--docker-image` in the same command.
+
+## Recommended / Default Docker Flow: `--base-docker-image` and `--script-dir`
+Pull in a local directory with scripts or files interested in running in
+`xpk workload create`.
+
+* `--base-docker-image` to set the base image that xpk will start with.
+* `--script-dir` to set which directory to pull into the image.
+* See `python3 xpk/xpk.py workload create --help` for more.
+
+Example with defaults which pulled the local directory into the base image:
+```shell
+echo -e '#!/bin/bash \n echo "Hello world from a test script!"' > test.sh
+# --script-dir is set by default to the current working directory.
+python3 xpk/xpk.py workload create --cluster xpk-test \
+--workload xpk-test-workload-base-image --command "bash test.sh" \
+--tpu-type=v5litepod-16 --num-slices=1 --base-docker-image=python:3.10
+```
+
+## Optional Direct Docker Image Configuration: `--docker-image`
+If a user wants to directly set the docker image used and not layer in the
+current working directory, set `--docker-image` to the image to be use in the
+workload.
+
+* Recommended Flow For Normal Sized Jobs (fewer than 10k accelerators)
```

Comment:
Sorry for sending you on a wild goose chase -- but this isn't really the recommended flow less than 10k accelerators. The recommended flow for small jobs is the autolayered approach. Can we call this "Running with ---docker-image"

Reply:
Done, and now understood!